AbstractSQLCompiler.ts

```diff
@@ -45,37 +45,48 @@ export interface Relationship {
 }
 export interface AbstractSqlQuery extends Array<AbstractSqlQuery | string> {}
 export interface AbstractSqlModel {
-	synonyms: {
-		[ synonym: string ]: string
-	}
-	relationships: {
-		[ resourceName: string ]: Relationship
-	}
-	tables: {
-		[ resourceName: string ]: AbstractSqlTable
-	}
+	synonyms: ResourceMap<string>
+	relationships: ResourceMap<Relationship>
+	tables: ResourceMap<AbstractSqlTable>
 	rules: AbstractSqlQuery[]
 }
 export interface SqlModel {
-	synonyms: {
-		[ synonym: string ]: string
-	}
-	relationships: {
-		[ resourceName: string ]: Relationship
-	}
-	tables: {
-		[ resourceName: string ]: AbstractSqlTable
-	}
+	synonyms: ResourceMap<string>
+	relationships: ResourceMap<Relationship>
+	tables: ResourceMap<AbstractSqlTable>
 	rules: SqlRule[]
 	createSchema: string[]
 	dropSchema: string[]
 }
 
+export interface hasDependants {
+	[dependant: string]: true
+}
+
+export interface schemaDependencyMap {
```

Comment:
Can you use `CamelCase` for these interfaces please, to distinguish them from vars